### ISSUE TITLE
🚧Fix request header issue

### ISSUE THREAD

**Issue by CalvinWilkinson** [KinsonDigital/GitHubData#37]
### Complete The Item Below

- [X] I have updated the title without removing the 🚧 emoji.

### Description

Fix an issue where each method call to the `GitHubDataService` clears the headers before adding the **Accept** header.

**Reason:**
The issue with this is that when a header is added in another application, simply calling any of the `GitHubDataService` methods will clear out the header that was just added.

**Solution:**
Remove the `ClearHeaders()` method call in all of the methods of the service.  Once this is complete, add the ability for the adding of a header to be checked to see if it already exists.  If it already exists, do not add it.  

Also, currently the headers are saved in an internal field that is the type `Dictionary<string, string>`.  This is a problem because traditionally, you can have multiple headers with the name **Accept** with different values.  This is also an issue.  Instead of a `Dictionary<string, string>`, use the type `List<(string name, string value)>` instead.

When checking if the header already exists, do not only go by the header name, but by the combination of the header name AND the value.   If both the header name AND value match, then that is the criteria to prevent the adding of the new header.

**⚠️WARNING⚠️**
This will be a https://github.com/KinsonDigital/GitHubData/labels/%F0%9F%A7%A8breaking%20changes because the behavior of the `GitHubDataService` class will be changing due to the removing of the `HttpClient.ClearHeaders()` method calls being removed.

### Acceptance Criteria

- [x] Remove the `ClearHeader()` method calls in all of the `GitHubDataService` methods.
  - [x] Adjust and update the unit tests
- [x] Replace the `HttpClient` internal header list of type `Dictionary<string, string>` with `List<(string name string value)>`
- [x] In the appropriate areas of the `HttpClient` class, check if the header exists by matching the name and the value.
  - If it already exists, just simply exit the method.

### ToDo Items

- [X] Change type labels added to this issue.  Refer to the _**Change Type Labels**_ section below.
- [X] Priority label added to this issue.  Refer to the _**Priority Type Labels**_ section below.
- [X] Issue linked to the correct project _(if required)_.
- [X] Issue linked to the correct milestone _(if required)_.
- [x] Draft pull request created and linked to this issue _(only required with code changes)_.

### Additional Information:

**_<details closed><summary>Change Type Labels</summary>_**

| Change Type         | Label                                                                                |
|---------------------|--------------------------------------------------------------------------------------|
| Bug Fixes           | https://github.com/KinsonDigital/GitHubData/labels/%F0%9F%90%9Bbug                     |
| Breaking Changes    | https://github.com/KinsonDigital/GitHubData/labels/%F0%9F%A7%A8breaking%20changes      |
| New Feature         | https://github.com/KinsonDigital/GitHubData/labels/%E2%9C%A8new%20feature              |
| Workflow Changes    | https://github.com/KinsonDigital/GitHubData/labels/workflow                            |
| Code Doc Changes    | https://github.com/KinsonDigital/GitHubData/labels/%F0%9F%93%91documentation%2Fcode    |
| Product Doc Changes | https://github.com/KinsonDigital/GitHubData/labels/%F0%9F%93%9Ddocumentation%2Fproduct |
</details>

**_<details closed><summary>Priority Type Labels</summary>_**

| Priority Type       | Label                                                              |
|---------------------|--------------------------------------------------------------------|
| Low Priority        | https://github.com/KinsonDigital/GitHubData/labels/low%20priority    |
| Medium Priority     | https://github.com/KinsonDigital/GitHubData/labels/medium%20priority |
| High Priority       | https://github.com/KinsonDigital/GitHubData/labels/high%20priority   |
</details>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

**Comment by CalvinWilkinson** [comment]
https://github.com/KinsonDigital/GitHubData/labels/%F0%9F%A7%A8breaking%20changes
It was discovered that the `HttpClient.AddOrUpdateHeader()` and `IHttpClient.AddOrUpdateHeader()` method had to be removed.  This is due to the fact that HTTP requests are allowed to have multiple headers with the same name.  This means that updating the header would be impossible because if you do not know which header to update, the method would not provided any value.

If you want to "update" a header, then that will have to be done manually in your consuming application by searching and removing them until they are all gone and then re-adding the header.

**Comment by CalvinWilkinson** [comment]
https://github.com/KinsonDigital/GitHubData/labels/%F0%9F%A7%A8breaking%20changes

Refactored the method `HttpClient` class and `IHttpClient` method names from `AddHeader()` and `AddHeaders()` to `AddRequestHeader()` and `AddRequestHeaders()`.

**Reason:**
This made it more clear that the headers being added are only for the next request that will be made and only for that request.
This makes it much more obvious about the purpose of the methods just like the `AddDefaultHeader()` and `AddDefaultHeaders()` methods which are used for every request.

**Comment by CalvinWilkinson** [comment]
https://github.com/KinsonDigital/GitHubData/labels/%E2%9C%A8new%20feature

Add 2 new properties to the `HttpClient` class and associated `IHttpClient` interface that returns the list of request headers and default headers.

**Reason:**
Some users will need to see what kinds of headers are added for the next request or for every request.   This will also let some users have the ability to manage the headers as well.
